voxel_grid/CMakeLists.txt

```diff
@@ -1,38 +1,36 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(voxel_grid)
 
-find_package(catkin REQUIRED
-  COMPONENTS
-    roscpp
-)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    voxel_grid
-  CATKIN_DEPENDS
-    roscpp
-)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
```

Comment:
Looking at the code, this package doesn't even use `rclcpp` anymore (it didn't really use `roscpp` other than for logging and static assert before). You could remove the dependency on `rclcpp` here (and in the `package.xml`) and either stick with `printf` for logging or just depend on `rcutils` for the logging macros. If you go for the former, you could also convert this to a plain cmake package, avoiding the dependency on `ament_cmake` also.